### PR TITLE
cleaning up BackendServiceException and Logging

### PR DESCRIPTION
**Pull request by saneshark**
Fixes https://github.com/department-of-veterans-affairs/vets.gov-team/issues/2897
Fixes https://github.com/department-of-veterans-affairs/vets.gov-team/issues/3772
Partially related to https://github.com/department-of-veterans-affairs/vets.gov-team/issues/3773

- removed icn and mhv_correlation_id logging from mhv_account.rb
- moved icn and mhv_correlation_id logging into application_controller
- removed all SentryLogging from BackendServiceException
- moved all logging for BackendServiceException into application_controller where additional context can easily be added.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -47,6 +47,14 @@ def routing_error
     # report the original 'cause' of the exception when present
     if SKIP_SENTRY_EXCEPTION_TYPES.include?(exception.class) == false
       extra = exception.respond_to?(:errors) ? { errors: exception.errors.map(&:to_hash) } : {}
+      if exception.is_a?(Common::Exceptions::BackendServiceException)
+        more_extra = { icn: current_user.try(:icn), mhv_correlation_id: current_user.try(:mhv_correlation_id) }
+        # Add additional user specific context to the logs
+        extra.merge!(more_extra) if current_user.present?
```
Comment (aub) [inline app/controllers/application_controller.rb]: Nitpicking but you could wrap both of these lines in the if test, remove the try calls and even remove the more_extra variable for a tiny cleanup

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -47,6 +47,14 @@ def routing_error
...
+        if exception.generic_error?
+          log_message_to_sentry(exception.va900_message, :warn, i18n_exception_hint: exception.va900_hint)
+        end
+      end
```
Comment (aub) [inline app/controllers/application_controller.rb]: I'm probably just being slow but why do we log a message and then the exception? Couldn't the message just be added to the exception?

**Comment by saneshark** [comment]
@markolson tagging you on this as well, since I think you guys used BackendServiceException. I wanted to make sure you don't mind that we'd add additional "extra content" like `icn` and `mhv_correlation_id` to the exception logging, even when it isn't relevant to you guys. You might have some thoughts on helpful content to add for HCA.

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (saneshark) [inline app/controllers/application_controller.rb]: warning about the va900 message is to encourage using exception.en.yml, I'm afraid if we don't have those annoying warnings we'll never add it. That being said, we've had those annoying warnings for quite sometime, and I only now got around to adding them as part of #1165

**Pull request merged**